service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NoticeOfChangeEventHandler.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.events.NoticeOfChangeEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.RespondentSolicitor;
+import uk.gov.hmcts.reform.fpl.model.notify.NotifyData;
+import uk.gov.hmcts.reform.fpl.service.email.NotificationService;
+import uk.gov.hmcts.reform.fpl.service.email.content.NoticeOfChangeContentProvider;
+
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.NOTICE_OF_CHANGE_FORMER_REPRESENTATIVE;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.NOTICE_OF_CHANGE_NEW_REPRESENTATIVE;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class NoticeOfChangeEventHandler {
+
+    private final NotificationService notificationService;
+    private final NoticeOfChangeContentProvider noticeOfChangeContentProvider;
+
+    @Async
+    @EventListener
+    public void notifySolicitorAccessGranted(final NoticeOfChangeEvent event) {
+        CaseData caseData = event.getCaseData();
+
+        RespondentSolicitor solicitor = event.getNewRespondentSolicitor();
+        String recipient = solicitor.getEmail();
+
+        NotifyData notifyData =
+            noticeOfChangeContentProvider.buildRespondentSolicitorAccessGrantedNotification(caseData,
+            solicitor);
+
+        notificationService.sendEmail(NOTICE_OF_CHANGE_NEW_REPRESENTATIVE, recipient, notifyData, caseData.getId());
+    }
+
+    @Async
+    @EventListener
+    public void notifySolicitorAccessRevoked(final NoticeOfChangeEvent event) {
+        CaseData caseData = event.getCaseData();
+
+        RespondentSolicitor solicitor = event.getOldRespondentSolicitor();
+        String recipient = solicitor.getEmail();
```

Comment:
in case solicitor is added via NoC (not changed) then i think prev solicitor is null (there was no solicitor at all for respondent). 
For sure previous solicitor email does not have to be present, and code should be ready to handle this (skip email sending, log fact that there is no prev solicitor or email is missing)